synapse/federation/federation_client.py

```diff
@@ -352,11 +366,11 @@ async def _record_failure_callback(
     @tag_args
     async def get_pdu(
         self,
-        destinations: Iterable[str],
+        destinations: Collection[str],
```

Comment:
Noticed that we iterate over `destinations` twice so if it was actually an `Iterable` anywhere it would get exhausted the first time we iterate in the log before we actually went over each destination.

Reply:
Re-commented on the diff https://github.com/matrix-org/synapse/pull/14096#discussion_r989731708